packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.js

```diff
@@ -4,80 +4,82 @@ import { interpolate } from '@visx/vendor/d3-interpolate'
+  return select(element)
     .data([value])
     .transition()
-    .duration(0)
-    .textTween(() => {
-      return () => {
-        setAnimated(true) // animation complete!
-        return formatValue(value)
-      }
-    })
-  }
+    .duration(duration)
+    .textTween(() => t => formatValue(interpolator(t)))
+    .end()
+}
 
-  function formatValue(num) {
-    return format(',d')(num)
-  }
+const initialValue = 0
 
-  useEffect(() => {
-    // If we already animated the number once, don't observe intersection
-    // This could happen if the value prop updates, but page did not refresh
-    if (animated) return
+function AnimatedNumber({ duration = 1000, value = 0 }) {
+  const numRef = useRef(null)
+  const [animated, setAnimated] = useState(false)
+  // render the value directly after the animation is complete.
+  const displayedValue = animated ? value : initialValue
 
+  useEffect(() => {
+    /*
+     * We're animating the DOM node content outside of the React render cycle
+     * so use node.textContent to check if we've already rendered the value.
+     */
     const numElement = numRef.current
+    const animationInProgress = numElement.textContent !== formatValue(initialValue)
+    if (animationInProgress) return // there's an animation already running.
+    if (animated) return // only run the intersection observer once.
+    if (formatValue(value) === numElement.textContent) return // nothing to animate yet.
 
-    const intersectionObserver = new window.IntersectionObserver(entries => {
+    const intersectionObserver = new window.IntersectionObserver(async entries => {
       // If intersectionRatio is 0, the target is out of view and we do not need to do anything.
       if (entries[0].intersectionRatio <= 0) return
 
-      // Once target element is in viewport, animate it then unobserve
-      if (!prefersReducedMotion && !animated) {
-        animateValue()
-        intersectionObserver.unobserve(numElement)
-      } else {
-        lessAnimation()
-        intersectionObserver.unobserve(numElement)
-      }
+      /*
+       * Once target element is in viewport, animate it then unobserve.
+       * Only run the full animation if prefers reduced motion is false
+       * and d3 hasn't already started an animation.
+       */
+      const animationInProgress = numElement.textContent !== formatValue(initialValue)
+      const animationDuration =
+        !prefersReducedMotion() && !animationInProgress ? duration : 0
+      intersectionObserver.unobserve(numElement)
+      await animateValue(numElement, initialValue, value, animationDuration)
+      setAnimated(true)
     })
 
     intersectionObserver.observe(numElement)
 
     return () => {
       intersectionObserver.disconnect()
     }
-  }, [numRef.current, animated])
+  }, [initialValue, value, duration])
 
-  return <span ref={numRef}>{!animated ? initialValue : formatValue(value)}</span>
+  return <span ref={numRef}>{formatValue(displayedValue)}</span>
```

Comment:
Note that if an animation is running, d3 will override `displayedValue` here with the output from the `textTween` function. If the `value` prop changes while d3 is animating the number, you want to wait for that animation to complete before updating `displayedValue`.